delta/plugins/blazegraph/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/blazegraph/indexing/BlazegraphIndexingCoordinator.scala

```diff
@@ -0,0 +1,181 @@
+import com.typesafe.scalalogging.Logger
+import fs2.Stream
+import monix.bio.Task
+import monix.execution.Scheduler
+import retry.CatsEffect._
+import retry.syntax.all._
+
+import java.util.Properties
+import scala.jdk.CollectionConverters._
+
+object BlazegraphIndexingCoordinator {
+
+  private val logger: Logger = Logger[BlazegraphIndexingCoordinator.type]
+
+  private val indexProperties: Map[String, String] = {
+    val props = new Properties()
+    props.load(getClass.getResourceAsStream("/blazegraph/index.properties"))
+    props.asScala.toMap
+  }
+
+  /**
+    * Create a coordinator for indexing Blazegraph view.
+    */
+  def apply(
+      views: BlazegraphViews,
+      eventLog: GlobalEventLog[Message[ResourceF[Graph]]],
+      client: BlazegraphClient,
+      projection: Projection[Unit],
+      config: BlazegraphViewsConfig
+  )(implicit
+      as: ActorSystem[Nothing],
+      scheduler: Scheduler,
+      base: BaseUri,
+      lens: ViewLens[ResourceF[IndexingBlazegraphView]]
+  ): Task[IndexingStreamCoordinator[ResourceF[IndexingBlazegraphView]]] = {
+
+    def buildStream(
+        viewRes: ResourceF[IndexingBlazegraphView],
+        initialProgress: ProjectionProgress[Unit]
+    ): Task[Stream[Task, Unit]] = {
+      val view                                = viewRes.value
+      implicit val projectionId: ProjectionId = viewRes.projectionId
+      val namespace                           = s"${config.client.indexPrefix}_${view.uuid}_${viewRes.rev}"
+      val retryStrategy                       = RetryStrategy[SparqlClientError](
+        config.client.retry,
+        {
+          case WrappedHttpClientError(_) => true
+          case _                         => false
+        },
+        logError(logger, "blazegraph client")
+      )
+      import retryStrategy.{errorHandler, policy, retryWhen}
+      for {
+        _     <- client
+                   .createNamespace(namespace, indexProperties)
+                   .retryingOnSomeErrors(retryWhen)
+                   .hideErrorsWith(err => new IllegalArgumentException(err.toString()))
+        eLog  <- eventLog
+                   .stream(view.project, initialProgress.offset, view.resourceTag)
+                   .hideErrorsWith(e => new IllegalArgumentException(e.reason))
+        stream = eLog
+                   .filterMessage(res =>
+                     (view.resourceTypes.isEmpty || res.types.exists(view.resourceTypes.contains))
+                       && (view.resourceSchemas.isEmpty || view.resourceSchemas.contains(
+                         res.schema.iri
+                       )) && (!res.deprecated || view.includeDeprecated)
+                   )
+                   .flatMapMessage { msg =>
+                     val graph = if (view.includeMetadata) {
+                       val types: Set[Triple] = msg.value.types.map((msg.value.id, rdf.tpe, _))
+                       msg.value.value
```

Comment:
I don't think all of this is necessary. You should just do:
```
msg.void.toNTriples
```
This is an IO, so when there is an error you should issue a `Message.failed` and when is a success just a successmessage as usually.